crates/ruff/src/checkers/noqa.rs

```diff
@@ -161,11 +161,16 @@ pub fn check_noqa(
                                     Location::new(row + 1, start - leading_spaces),
                                     Location::new(row + 1, end + trailing_spaces),
                                 ));
-                            } else {
+                            } else if lines[row].chars().nth(end + 1).map_or(false, |c| c == '#') {
```

Comment:
Let's use `skip_while(char::is_whitespace)` or call `lines[row][end...].trim_start()` to handle  `# noqua     # comment` correctly.


Can you also add a test case for it.

Reply:
Will do, added test.